### PR TITLE
Switch away from os.distro.codename fact

### PR DESCRIPTION
**Pull request by root-expert**
#### Pull Request (PR) description
Releated info: https://github.com/voxpupuli/beaker-hostgenerator/pull/244

### CODE REVIEW

File: manifests/params.pp
Comment (ekohl) [inline manifests/params.pp:115]: Slightly surprised this works. I'd expect it to be case sensitive.
```suggestion
      if ($facts['os']['name'] == 'Ubuntu' and $facts['os']['release']['major'] == '16.04') {
```

**Review by ekohl — APPROVED** [review]

File: manifests/params.pp
Comment (root-expert) [reply manifests/params.pp]: Maybe it doesn't and our CI doesn't catch it :smile:

**Comment by root-expert** [comment]
I guess we have to wait for my PR on puppetlabs-apt to get merged before merging this.

**Comment by kenyon** [comment]
Anything related to EoL Ubuntu 16.04 could also just be removed I think.

File: manifests/params.pp
Comment (smortex) [reply manifests/params.pp]: Puppet string comparison is case-insensitive:

https://puppet.com/docs/puppet/7/lang_expressions.html#comparison-equality

**Review by ekohl — APPROVED** [review]
Thanks!

**Comment by ekohl** [comment]
> Anything related to EoL Ubuntu 16.04 could also just be removed I think.

In the short term, this improves things without having to do a major release.

**Pull request merged**
